docs/weave-tutorial/05-messages.md

```diff
@@ -0,0 +1,109 @@
+
+## Defining Messages
+
+Messages are similar to the `POST` endpoints in a typical API. They are the only way to effect a change in the system. Ignoring the issue of authentication and rate limitation, which is handled by the Decorators / Middleware, when we design Messages, we focus on all possible state transitions and the information they need to proceed.
+
+In the Orderbook example, we can imagine:
+
+- Create market
+- Create order book
+- Create order
+
+## Dive into Code
+
+First create your `msg.go` file. This is where the magic will happen.
+
+First create an __init()__ function and register to migration schema:
+
+```go
+func init() {
+    // Migration needs to be registered for every message introduced in the codec.
+    migration.MustRegister(1, &CreateOrderBookMsg{}, migration.NoModification)
+}
+```
+
+Define path that will be used for routing messages to handler:
+
+```go
+func (CreateOrderBookMsg) Path() string {
+    return "orderbook/create_orderbook"
+}
+```
+
+After that ensure your message is a `weave.Msg`:
+
+```go
+var _ weave.Msg = (*CreateOrderBookMsg)(nil)
+```
+
+## Validation
+
+While validation for data models is much more like SQL constraints: “max length 20”, “not null”, “constaint foo > 3”, validation for messages is validating potentially malicious data coming in from external sources and should be validated more thoroughly. One may want to use regexp to avoid control characters or null bytes in a “string” input. Maybe restrict it to alphanumeric or ascii characters, strip out html, or allow full utf-8. Addresses must be checked to be the valid length. Amount being sent to be positive (else I send you -5 ETH and we have a **TakeMsg**, instead of **SendMsg**).
+
+The validation on Messages should be a lot more thorough and well tested than the validation on data models, which is as much documentation of acceptable values as it is runtime security.
+
+`Validate` method of `CreateOrderBookMsg`:
+
+```go
+func (m CreateOrderBookMsg) Validate() error {
+    if err := m.Metadata.Validate(); err != nil {
+        return errors.Wrap(err, "metadata")
+    }
+    if err := validID(m.MarketID); err != nil {
+        return errors.Wrap(err, "market id")
+    }
+    if !coin.IsCC(m.AskTicker) {
+        return errors.Wrapf(errors.ErrCurrency, "Invalid Ask Ticker: %s", m.AskTicker)
+    }
+    if !coin.IsCC(m.BidTicker) {
+        return errors.Wrapf(errors.ErrCurrency, "Invalid Bid Ticker: %s", m.BidTicker)
+    }
+    if m.BidTicker <= m.AskTicker {
+        return errors.Wrapf(errors.ErrCurrency, "ask (%s) must be before bid (%s)", m.AskTicker, m.BidTicker)
+    }
+    return nil
+}
+```
+
+```go
+func validID(id []byte) error {
```

Comment:
I think `validateID` might be a better name. `validID` sounds like this function returns a `bool` result.
Adding a comment might be nice touch :nail_care:

Reply:
On it. I will apply the same change on tutorial then here